BJS2-73242 Spotify lock

dextomi opened the pull request:
task completed!

dextomi committed:
Spotify lock

dextomi committed:
removed empty lines

MarselYsup commented on src/main/java/school/faang/spotify_lock/Player.java:
```diff
@@ -0,0 +1,99 @@
+            try {
+                Thread.sleep(10);
+            } catch (InterruptedException e) {
+                throw new RuntimeException(e);
```
тут нужно добавить 
```java
Thread.currentThread().interrupt()
```
 , чтобы установить флаг прерывания текущего потока. Это позволяет потоку узнать, что его просят завершить работу или прервать текущую операцию.

MarselYsup replied:
Лучше использовать конкретного наследника RuntimeException, например, IllegalStateException. Всегда передавай сообщение с информацией о контексте исключения.

MarselYsup commented on src/main/java/school/faang/spotify_lock/Player.java:
```diff
@@ -0,0 +1,99 @@
+    public void play() {
+        synchronized (lock) {
+            try {
+                Thread.sleep(10);
```
лучше вынести магическое число в константу

MarselYsup commented on src/main/java/school/faang/spotify_lock/Player.java:
```diff
@@ -0,0 +1,99 @@
+    public void pause() {
+        synchronized (lock) {
+            try {
+                Thread.sleep(10);
```
Можно вынести в константу

MarselYsup commented on src/main/java/school/faang/spotify_lock/Player.java:
```diff
@@ -0,0 +1,99 @@
+            try {
+                Thread.sleep(10);
+            } catch (InterruptedException e) {
+                throw new RuntimeException(e);
```
тут нужно добавить 
```java
Thread.currentThread().interrupt()
````
 , чтобы установить флаг прерывания текущего потока. Это позволяет потоку узнать, что его просят завершить работу или прервать текущую операцию.

MarselYsup commented on src/main/java/school/faang/spotify_lock/Player.java:
```diff
@@ -0,0 +1,99 @@
+    public void skip() {
+        synchronized (lock) {
+            try {
+                Thread.sleep(10);
```
лучше вынести в константу

MarselYsup commented on src/main/java/school/faang/spotify_lock/Player.java:
```diff
@@ -0,0 +1,99 @@
+            try {
+                Thread.sleep(10);
+            } catch (InterruptedException e) {
+                throw new RuntimeException(e);
```
тут нужно добавить 
```java
Thread.currentThread().interrupt()
```
 , чтобы установить флаг прерывания текущего потока. Это позволяет потоку узнать, что его просят завершить работу или прервать текущую операцию.

MarselYsup commented on src/main/java/school/faang/spotify_lock/Player.java:
```diff
@@ -0,0 +1,99 @@
+    public void previous() {
+        synchronized (lock) {
+            try {
+                Thread.sleep(10);
```
Добавь в константу

MarselYsup commented on src/main/java/school/faang/spotify_lock/Player.java:
```diff
@@ -0,0 +1,99 @@
+            try {
+                Thread.sleep(10);
+            } catch (InterruptedException e) {
+                throw new RuntimeException(e);
```
тут нужно добавить 
```java
Thread.currentThread().interrupt()
```
 , чтобы установить флаг прерывания текущего потока. Это позволяет потоку узнать, что его просят завершить работу или прервать текущую операцию.

MarselYsup commented on src/main/java/school/faang/spotify_lock/Player.java:
```diff
@@ -0,0 +1,99 @@
+                case NEXT -> {
+                    if (currentTrackIndex < playList.size() - 1) {
+                        currentTrack = playList.get(currentTrackIndex + 1);
+                    }
```
можно добавлять информацию о том, что дальше трека нету и в PREVIOUS аналогично

MarselYsup reviewed (commented):
Интересная задумка со switch 👍
Поправь несколько комментариев, а так по логике все хорошо 👍

dextomi committed:
Перенес логику в отдельные методы и сделал их приватными;
Добавил енамы:
Переименовал changeTrack в trackAction, сделал его публичным и вызов остальных методов также сделал в нем через switch case.
